tests/rules/property.js

```diff
@@ -664,5 +673,35 @@ eslintTester.run("property", rule, {
             ],
             parser: require.resolve("../parsers/fantasy-operator"),
         },
+
+        // Uninitialized variable declaration.
+        {
+            code: `
+              let c;
+              if (cond) {
+                c = '<b>safe</b>';
+              } else {
+                c = evil;
+              }
+              a.innerHTML = \`\${c}\`;
+            `,
+            errors: [
+                {
+                    message: /Unsafe assignment to innerHTML/,
+                    type: "AssignmentExpression"
+                }
+            ],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "let c; c = 'apparently-safe'; functionCall(c); n.innerHTML = c.property;",
+            parserOptions: { ecmaVersion: 6 },
+            errors: [
+                {
+                    message: /Unsafe assignment to innerHTML/,
+                    type: "AssignmentExpression"
+                }
+            ],
+        },
```

Comment:
> I don't think this test is doing what we intended.
The linter is probably going to treat c and c.property like two completely separate variables (like foo and bar).
I don't think the functionCall would be treated like a potentially-writing reference.
>
> @rpl You suggested this, right? Can you explain what you had in mind?

@mozfreddyb yes, with the current implementation I would expect that `c` and `c.property` would be considered two separate variables and the assignment `n.innerHTML = c.property` to be considered unsafe.

My idea/concern was to ensure that would still be the case even if we change the implementation with more than a few small tweaks (e.g. like if we would be refactoring the entire rule and/or ruleHelpers module), basically a smoke test to be sure that the variable tracking implementation (independently from how it is actually implemented) isn't going to change that expectation.

I'm definitely ok to proceed without that test case if we are unsure about its actual value, it may be just me being paranoid :-P

Reply:
Gotcha. let's leave it in :)